### PR TITLE
applications: nrf5340_audio: bt_stream cleanup

### PR DESCRIPTION
**Pull request by erikrobstad**
Cleanup when looking at
the bt_stream module implementation.

### CODE REVIEW

File: applications/nrf5340_audio/src/bluetooth/Kconfig [hunk 1]
```diff
@@ -21,6 +18,11 @@ if TRANSPORT_CIS
 rsource "bt_stream/unicast/Kconfig"
 endif # TRANSPORT_CIS
 
+menu "Bluetooth"
```
Comment (alexsven) [inline applications/nrf5340_audio/src/bluetooth/Kconfig]: The reason the Bluetooth was above rsource is because we want renderer, content_ctrl etc all should be under the menu-tab ``Bluetooth``, so I suggest it is moved up again, same with ``Bluetooth audio``

Comment (alexsven) [reply applications/nrf5340_audio/src/bluetooth/Kconfig]: I see that his is a global change. The order of menus and rsource does matter, so they shouldn't necessarily be moved to the top

File: applications/nrf5340_audio/src/bluetooth/bt_management/controller_config/Kconfig.defaults [hunk 2]
```diff
@@ -15,4 +15,8 @@ config BT_ATT_ENFORCE_FLOW
 config BT_HCI_VS_EXT
 	default n
 
+# Disable HCI ACL flow control
+config BT_HCI_ACL_FLOW_CONTROL
```
Comment (alexsven) [inline applications/nrf5340_audio/src/bluetooth/bt_management/controller_config/Kconfig.defaults:19]: Why was this added? Was this disabled before as well?

File: applications/nrf5340_audio/src/bluetooth/bt_stream/broadcast/broadcast_sink.c [hunk 3]
```diff
@@ -59,17 +59,19 @@ static struct active_audio_stream active_stream;
 static uint8_t sync_stream_cnt;
 static uint8_t active_stream_index;
 
-/* We need to set a location as a pre-compile, this changed in initialize */
+/* We need to set a location as a pre-compile, this changed when initializing */
```
Comment (alexsven) [inline applications/nrf5340_audio/src/bluetooth/bt_stream/broadcast/broadcast_sink.c]:
```suggestion
/* We need to set a location as a pre-compile, this changes when initializing */
```

File: applications/nrf5340_audio/src/bluetooth/bt_stream/broadcast/broadcast_source.c [hunk 4]
```diff
@@ -168,11 +207,11 @@ static void stream_stopped_cb(struct bt_bap_stream *stream, uint8_t reason)
 }
 
 static struct bt_bap_stream_ops stream_ops = {
-	.started = stream_started_cb,
-	.stopped = stream_stopped_cb,
 #if (CONFIG_BT_AUDIO_TX)
```
Comment (alexsven) [inline applications/nrf5340_audio/src/bluetooth/bt_stream/broadcast/broadcast_source.c]: Shouldn't need this guard, same on line:150. If we are broadcast_source then BT_RX should be set. I think this was left from when we compiled all files all the time. We shouldn't need to compile this file unless broadcast_source is set

File: applications/nrf5340_audio/src/modules/button_handler.c [hunk 5]
```diff
@@ -22,43 +22,40 @@
...
+			},
+			{
+				.btn_name = STRINGIFY(BUTTON_4), .btn_pin = BUTTON_4,
+						      .btn_cfg_mask =
+							      DT_GPIO_FLAGS(DT_ALIAS(sw3), gpios),
+				},
+				{
+					.btn_name = STRINGIFY(BUTTON_5), .btn_pin = BUTTON_5,
+						.btn_cfg_mask = DT_GPIO_FLAGS(DT_ALIAS(sw4), gpios),
+					}};
```
Comment (alexsven) [inline applications/nrf5340_audio/src/modules/button_handler.c]: This now looks like shit, at least here in github, maybe we should see if there is anything we can do about it. (Adding commas or something similar)

File: applications/nrf5340_audio/src/bluetooth/bt_stream/unicast/unicast_server.c [hunk 6]
```diff
@@ -501,7 +501,7 @@ int unicast_server_adv_populate(struct bt_data *adv_buf, uint8_t adv_buf_vacant)
 
 #if defined(CONFIG_BT_CSIP_SET_MEMBER)
 	ret = adv_buf_put(adv_buf, adv_buf_vacant, &adv_buf_cnt, BT_DATA_CSIS_RSI,
-			  ARRAY_SIZE(csis_rsi_adv_data), &csis_rsi_adv_data[0]);
+			  ARRAY_SIZE(csip_rsi_adv_data), &csip_rsi_adv_data[0]);
```
Comment (alexsven) [inline applications/nrf5340_audio/src/bluetooth/bt_stream/unicast/unicast_server.c]: The type that we put in this LTV is BT_DATA_CSIS_RSI, so I think we should keep the csis_rsi_adv_data and remove the csip_rsi_adv_data.

**Review by alexsven — DISMISSED** [review]

File: applications/nrf5340_audio/src/audio/streamctrl_broadcast_source.c [hunk 7]
```diff
@@ -321,7 +321,7 @@ void streamctrl_send(void const *const data, size_t size, uint8_t num_ch)
 	int ret;
 	static int prev_ret;
 
-	struct encoded_audio enc_audio = {.data = data, .size = size, .num_ch = num_ch};
+	struct le_audio_encoded_audio enc_audio = {.data = data, .size = size, .num_ch = num_ch};
```
Comment (koffes) [inline applications/nrf5340_audio/src/audio/streamctrl_broadcast_source.c:323]: This name I find too elaborate. We could send this data over WiFi.

File: applications/nrf5340_audio/src/bluetooth/bt_management/controller_config/Kconfig.defaults (hunk 2 above)
Comment (koffes) [reply applications/nrf5340_audio/src/bluetooth/bt_management/controller_config/Kconfig.defaults:19]: No, this fixes the flow control warning.

Comment (koffes) [reply applications/nrf5340_audio/src/bluetooth/bt_management/controller_config/Kconfig.defaults:19]: Recommend a comment on the lines of "flow control not supported on BT_LL_ACS_NRF53

File: applications/nrf5340_audio/src/bluetooth/bt_stream/broadcast/broadcast_source.c (hunk 4 above)
Comment (koffes) [reply applications/nrf5340_audio/src/bluetooth/bt_stream/broadcast/broadcast_source.c]: Agree

File: applications/nrf5340_audio/src/bluetooth/bt_stream/unicast/unicast_server.c [hunk 8]
```diff
@@ -345,13 +343,7 @@ static const struct bt_bap_unicast_server_cb unicast_server_cb = {
...
 };
 
-#if (CONFIG_BT_AUDIO_TX)
-static void stream_sent_cb(struct bt_bap_stream *stream)
-{
-	atomic_dec(&iso_tx_pool_alloc);
-}
-#endif /* (CONFIG_BT_AUDIO_TX) */
-
+#if (CONFIG_BT_AUDIO_RX)
```
Comment (koffes) [inline applications/nrf5340_audio/src/bluetooth/bt_stream/unicast/unicast_server.c:346]: We have been focusing on getting rid of precompile guards. Can we solve this differently?

File: applications/nrf5340_audio/src/bluetooth/bt_stream/unicast/unicast_server.h [hunk 9]
```diff
@@ -79,6 +69,6 @@ int unicast_server_disable(void);
  *
  * @return	0 for success, error otherwise.
  */
-int unicast_server_enable(le_audio_receive_cb rx_cb, timestamp_cb timestmp_cb);
+int unicast_server_enable(le_audio_receive_cb rx_cb);
```
Comment (koffes) [inline applications/nrf5340_audio/src/bluetooth/bt_stream/unicast/unicast_server.h:71]: Discuss: This was moved to zbus?

File: applications/nrf5340_audio/src/modules/button_handler.c (hunk 5 above)
Comment (koffes) [reply applications/nrf5340_audio/src/modules/button_handler.c]: This area is mauled by the formatter. I had to save without formatting.

**Review by koffes — CHANGES_REQUESTED** [review]

File: applications/nrf5340_audio/src/bluetooth/bt_management/controller_config/Kconfig.defaults (hunk 2 above)
Comment (alexsven) [reply applications/nrf5340_audio/src/bluetooth/bt_management/controller_config/Kconfig.defaults:19]: Then it should probably also be:
```suggestion
config BT_HCI_ACL_FLOW_CONTROL
default n if BT_LL_ACS_NRF53
```

File: applications/nrf5340_audio/src/bluetooth/bt_stream/unicast/unicast_server.c (hunk 8 above)
Comment (alexsven) [reply applications/nrf5340_audio/src/bluetooth/bt_stream/unicast/unicast_server.c:346]: Unfortunately no, the callbacks are inside compile dependent code in host. So the sent callback doesn't exist unless BT_AUDIO_RX is set, similarly with recv, but for BT_AUDIO_TX

File: applications/nrf5340_audio/src/bluetooth/bt_stream/unicast/unicast_server.h (hunk 9 above)
Comment (alexsven) [reply applications/nrf5340_audio/src/bluetooth/bt_stream/unicast/unicast_server.h:71]: yes

File: applications/nrf5340_audio/src/bluetooth/Kconfig (hunk 1 above)
Comment (erikrobstad) [reply applications/nrf5340_audio/src/bluetooth/Kconfig]: I did not realize this. Reverted back these changes

File: applications/nrf5340_audio/src/bluetooth/bt_management/controller_config/Kconfig.defaults (hunk 2 above)
Comment (erikrobstad) [reply applications/nrf5340_audio/src/bluetooth/bt_management/controller_config/Kconfig.defaults:19]: Changed the comment. And these configs are already inside if BT_LL_ACS_NRF53 (on line 8)

File: applications/nrf5340_audio/src/bluetooth/bt_stream/broadcast/broadcast_sink.c (hunk 3 above)
Comment (erikrobstad) [reply applications/nrf5340_audio/src/bluetooth/bt_stream/broadcast/broadcast_sink.c]: Done

File: applications/nrf5340_audio/src/bluetooth/bt_stream/broadcast/broadcast_source.c (hunk 4 above)
Comment (erikrobstad) [reply applications/nrf5340_audio/src/bluetooth/bt_stream/broadcast/broadcast_source.c]: Agreed, removed

File: applications/nrf5340_audio/src/modules/button_handler.c (hunk 5 above)
Comment (erikrobstad) [reply applications/nrf5340_audio/src/modules/button_handler.c]: Changed back and added /* clang-format off */

File: applications/nrf5340_audio/src/bluetooth/bt_stream/unicast/unicast_server.c (hunk 6 above)
Comment (erikrobstad) [reply applications/nrf5340_audio/src/bluetooth/bt_stream/unicast/unicast_server.c]: As we discussed in the Teams call, the name could be csip_rsi_adv_data as well, so keeping it for now.

File: applications/nrf5340_audio/src/audio/streamctrl_broadcast_source.c (hunk 7 above)
Comment (erikrobstad) [reply applications/nrf5340_audio/src/audio/streamctrl_broadcast_source.c:323]: That is true. But this struct is defined in le_audio.h, and it will only be used for LE Audio.
And as I understand it, it is a general naming scheme in le_audio.h to start with "le_audio"

File: applications/nrf5340_audio/src/bluetooth/bt_stream/unicast/unicast_server.c (hunk 8 above)
Comment (erikrobstad) [reply applications/nrf5340_audio/src/bluetooth/bt_stream/unicast/unicast_server.c:346]: I found 2 other places that #if could be removed

**Review by koffes — APPROVED** [review]

**Comment by erikrobstad** [comment]
Rebased after the unicast client PR was merged
